txnsync/profiler.go

```diff
@@ -46,6 +46,24 @@ const (
 	profFirstDetachedElement = profElementGetTxnsGroups
 )
 
+// We want to create this so we can test if we actually send metrics or not
+type iProfilerMetricLogger interface {
```

Comment:
I don't think this is needed. 
logging.Logger is already an interface. 

Look at logging/telemetry_test.go for telemetryTestFixture. 
My guess is similar fixture can be used here which will do the trick without having to modify the source.